### DISCUSSION TITLE
124.000 Flash writes

### DISCUSSION THREAD

**Discussion by cruiser2** [arendst/Tasmota#17589] [Q&A] [answered]
Hi
During about 2 months of continuous operation my Sonoff POW R2 accumulated more than 124.000 flash writes due to a Wifi configuration with an alternative Access Point (AP2) which did not work at the point of installation and AP1 switching OFF during the nights.
Program Version	12.1.1(tasmota)
Build Date & Time	2022-08-25T11:33:55
Core/SDK Version	2_7_4_9/2.2.2-dev(38a443e)
Uptime	0T04:19:54
Flash write Count	124083 at 0xF9000
Boot Count	107 (Reboot every day once with a Rule)
WifiConfig 4
By removing AP2 from Wifi Configuration I steadied the Flash writes but I think it is time to change the flash chip and here I request some help. Do I need to preflash the new chip before soldering? If so – how? If I can do the flashing after soldering, will Webinstaller do the job?
Thanks

**Comment by sfromis** [comment, marked as answer]
If you want to replace the flash chip, you can preflash if you want, and already have the tools. You do not need to, and after soldering you can simply treat it as a new device following the usual steps in https://tasmota.github.io/docs/Getting-Started/ - including tool recommendations.

Note that flash chip endurance is hard to predict, and it may not be close to croaking. Of course, I can't say that it is not, either....

**Reply by cruiser2** [reply]
Thanks for the fast reply. So I will simply solder a new chip in place and do a Web Installer job.
By the way, the datasheet says “More than 100K erase/program cycles”. So a chip replacement seems to be adequate for a unit which should work for some months without physical access.

**Reply by sfromis** [reply]
The question may still be how "erase/program cycles" exactly applies to updating settings. By default Tasmota will rotate writes to 8 slots, which in principle would be 100 K cycles per slot, assuming that these can really be counted as separate. Anyway, obviously "without physical access" (which I'd aim to avoid) is an argument for "better safe than sorry"....

**Reply by s-hadinger** [reply]
Indeed there are 8 slots. So it wrote ~15k in each slot. Chances are your device can still be used for a long time

**Reply by cruiser2** [reply]
I agree that my decision to change the flash was not justified in light of the implemented “wear leveling” which I did not consider. Given the actual numbers which do apply (about 15000 vs >100000) the POW R2 seems not to be at a higher risk by flash failure than of failure on any other component.

**Reply by Jason2866** [reply]
The power supply has a way more higher chance to fail, than the flash chip.

**Reply by sfromis** [reply]
Until now, I've at least not seen any of my Sonoff Pow R2 devices croak. Don't know when I get to that point :wink: 
OTOH, I've had two Sonoff Basic suddenly no longer work...

**Reply by Jason2866** [reply]
My Sonoff Pow (the first available, so it is old) still working and running development build from this week. Flash Id 1640E. That is a good winbond chip
No relay switched, just measuring 24/7.

**Comment by ortegafernando** [comment]
HI, this remember me to:

https://github.com/arendst/Tasmota/pull/15181
https://github.com/arendst/Tasmota/pull/14822

Could you add, please, a bit more information in INFO3 message to avoid this cases? We could monitor this flash count in nodered or similar to detect these cases.

what do you think @cruiser2 ?

Regads,

**Reply by sfromis** [reply]
If you want to monitor such details, you can just react to the initial MQTT messages from Tasmota and send a `Status 1` command for available details like `SaveCount`. Beware that this is close to **useless** for monitoring flash endurance, as counts can easily be reset, also without specific action to do that.

Anyway, INFO payloads are not really meant for such low level details....

BTW.... What I do is to react to the connect with a `Status0` command, providing a long message with all the Status info, in case I want to check something available there, using my message archive.
